pkg/envvar/env_var.go

```diff
@@ -0,0 +1,73 @@
+// envvar package helps converting name/value pairs to devfile EnvVars
+package envvar
+
+import (
+	"errors"
+	"strings"
+
+	devfilev1 "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+)
+
+// EnvVar represents an environment variable
+type EnvVar struct {
+	Name  string `yaml:"Name"`
+	Value string `yaml:"Value"`
+}
+
+// List represents a list of environment variables
+type List []EnvVar
+
+// ToDevfileEnv converts the EnvVarList to the list of Envs supported by devfile
+func (evl List) ToDevfileEnv() []devfilev1.EnvVar {
+	var envList []devfilev1.EnvVar
+	for _, ev := range evl {
+		envList = append(envList, devfilev1.EnvVar{
+			Name:  ev.Name,
+			Value: ev.Value,
+		})
+	}
+	return envList
+}
+
+// newFromString takes a string of format "name=value" and returns an Env
+// variable struct
+func newFromString(envStr string) (EnvVar, error) {
+	envList := strings.SplitN(envStr, "=", 2)
```

Comment:
:100:
I hadn't understood the use of `SplitN` until now.

Reply:
But using this also means that env vars such as `a==b` will split into [a, =b] or `a!=b` will split into [a!, b]. I don't think these can be considered as valid cases, can they? I understand these are corner cases and not likely to be encountered, but do you think we should cover them?

Reply:
For the first one, a = '=b', it is a valid case for me. We don't have to limit the possible values to pass to the container, the value '=b' seems completely legit to me.
For the second one, a! = b, I'm not sure that a! is a valid name for an environment variable. Looking more precisely to it, it is not valid: https://www.tutorialspoint.com/unix/unix-using-variables.htm